### PR TITLE
Feat/clean mv layout input

### PR DESCRIPTION
**Pull request by malmen237**
# What does this do?

## New default layout
On start the default layout is prepopulated with the sources from the production-setup. The slots that have not been populated is returned empty. 
Screenshot 2024-10-25 at 10 13 37

## Possible to have a completely clean layout
When box is ticked the layout is cleaned, when unclicked again the default layout is returned into view
Screenshot 2024-10-25 at 10 13 53

## Bug-fixes
- When removing sources on running production the multiview will return an empty view with empty label. ID is saved so if the source is added again it will reappear in multiview.
- Restructure of how multiview is updated both on production start and on production.active

### CODE REVIEW

File: src/components/modal/configureMultiviewModal/MultiviewLayoutSettings/RemoveLayoutButton.tsx [hunk 1]
```diff
@@ -1,31 +1,40 @@
 import { IconTrash } from '@tabler/icons-react';
+import { hi } from 'date-fns/locale';
```
Comment (Saelmala) [inline src/components/modal/configureMultiviewModal/MultiviewLayoutSettings/RemoveLayoutButton.tsx:2]: unused import

File: src/components/startProduction/StartProductionButton.tsx [hunk 2]
```diff
@@ -39,9 +40,24 @@ export function StartProductionButton({
   const putProduction = usePutProduction();
   const [stopProduction, loadingStopProduction] = useStopProduction();
   const [deleteMonitoring] = useDeleteMonitoring();
+  const [updateSourceInputSlotOnMultiviewLayouts, updateLading] =
```
Comment (Saelmala) [inline src/components/startProduction/StartProductionButton.tsx:43]: updateLading --> updateLoading

File: src/hooks/useCleanedDefaultPreset.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  sourceList,
+  isChecked
+}: {
+  sourceList: SourceReference[] | undefined;
+  isChecked: boolean;
+}) {
+  const [cleanedDefaultPresets, setCleanedDefaultPresets] = useState<
+    MultiviewPreset[]
+  >([]);
+  const [dirtyMultiviewPresets] = useMultiviewPresets();
```
Comment (Saelmala) [inline src/hooks/useCleanedDefaultPreset.tsx:16]: maybe give this a more descriptive name if possible

File: src/hooks/useUpdateSourceInputSlotOnMultiviewLayouts.tsx [hunk 4]
```diff
@@ -17,13 +17,13 @@ export function useUpdateSourceInputSlotOnMultiviewLayouts(): CallbackHook<
   const putProduction = usePutProduction();
 
   const updateSourceInputSlot = async (production: Production) => {
+    setLoading(true);
     const layouts = await multiviewLayouts();
     if (layouts) {
-      for (const singleLayout of layouts) {
+      const asd = layouts.map(async (singleLayout) => {
```
Comment (Saelmala) [inline src/hooks/useUpdateSourceInputSlotOnMultiviewLayouts.tsx:23]: asd=

File: src/components/modal/configureMultiviewModal/MultiviewLayoutSettings/MultiviewLayoutSettings.tsx [hunk 5]
```diff
@@ -95,6 +109,14 @@ export default function MultiviewLayoutSettings({
     }
   }, [layoutModalOpen]);
 
+  useEffect(() => {
+    if (layoutModalOpen) {
+      setRefresh(true);
+    } else {
+      setRefresh(false);
+    }
```
Comment (LucasMaupin) [inline src/components/modal/configureMultiviewModal/MultiviewLayoutSettings/MultiviewLayoutSettings.tsx:117]:
```suggestion
      setRefresh(layoutModalOpen);
```

File: src/components/modal/configureMultiviewModal/MultiviewLayoutSettings/MultiviewLayoutSettings.tsx [hunk 6]
```diff
@@ -163,26 +186,19 @@ export default function MultiviewLayoutSettings({
...
     if (layoutToRemove && !layoutToRemove._id) {
       toast.error(t('preset.could_not_delete_layout'));
       return;
     }
     if (layoutToRemove && layoutToRemove._id) {
       deleteLayout(layoutToRemove._id.toString()).then(() => {
         setRefresh(true);
-        if (multiviewPresets && multiviewPresets[0]) {
-          setSelectedMultiviewPreset(multiviewPresets[0]);
+        if (cleanedDefaultPresets && cleanedDefaultPresets[0]) {
```
Comment (LucasMaupin) [inline src/components/modal/configureMultiviewModal/MultiviewLayoutSettings/MultiviewLayoutSettings.tsx:200]:
```suggestion
        if (cleanedDefaultPresets?.[0]) {
```

**Review by LucasMaupin — APPROVED** [review]

**Review by Saelmala — CHANGES_REQUESTED** [review]
Left a few minor comments, but some I think are more important, eg. variable name asd

**Review by Saelmala — APPROVED** [review]

**Pull request merged**